src/store/reducers.js

```diff
@@ -1,58 +1,110 @@
 import * as actions from './actions'
+import { combineReducers } from 'redux'
 
-// ===================================================================
+import initialState from './initialStoreState'
+/* Reducers are synchronous and repeatable : it's not the place to put api call
+*/
 
-// import { combineReducers } from 'redux'
+function sessionReducer (currentSession = initialState.session, action) {
+  switch (action.type) {
+    case actions.SESSION_PATCH :
+      let payload = action.payload
+      // don't change loggued state from this action
+      delete payload.isLoggingIn
+      delete payload.isLoggingOut
+      delete payload.isLoggued
+      return Object.assign(
+        {},
+        currentSession,
+        payload // login , password
+      )
+    case actions.SIGN_IN :
+      // user tried to sign in
+      return Object.assign(
+        {},
+        currentSession,
+        action.payload, // login , password
+        {isLoggingIn: true}
+      )
 
-const createAsyncHandler = ({ error, next }) => (state, action) => {
-  if (action.error) {
-    if (error) {
-      return error(state, action.payload)
-    }
-  } else {
-    if (next) {
-      return next(state, action.payload)
-    }
-  }
-
-  return state
-}
+    case actions.SIGN_OUT :
+      return Object.assign({}, currentSession, {isLoggingOut: true})
 
-// Action handlers are reducers but bound to a specific action.
-const combineActionHandlers = (initialState, handlers) => {
-  for (const action in handlers) {
-    const handler = handlers[action]
-    if (typeof handler === 'object') {
-      handlers[action] = createAsyncHandler(handler)
-    }
-  }
+    case actions.SIGNED_OUT:
+      // when user is signed out : reset session
+      return initialState.session
 
-  return (state = initialState, action) => {
-    const handler = handlers[action.type]
+    case actions.SIGNED_IN :
+      // when user is signed in : session informations are received from server
+      return Object.assign(
+        {},
+        currentSession,
+        action.payload,
+        {
+          isLoggued: true,
+          isLoggingIn: false,
+          isLoggingOut: false,
+          password: 'redacted , don t need it '
+        }
+      )
 
-    return handler
-      ? handler(state, action)
-      : state
+    default :
+      return currentSession // reducer HAVE TO return a state, even when they did nothing
   }
 }
```

Comment:
You should use this function which simplifies the writing of reducers (no more `switch/case`) and preserve us from making trivial mistakes such as forgetting to return the state as default case.